_search-plugins/searching-data/inner-hits.md

```diff
@@ -0,0 +1,800 @@
+                      "parent" : "1"
+                    },
+                    "comments" : [
+                      {
+                        "user" : "John",
+                        "message" : "This is a comment"
+                      },
+                      {
+                        "user" : "Jane",
+                        "message" : "Another comment"
+                      }
+                    ]
+                  },
+                  "inner_hits" : {
+                    "comments" : {
+                      "hits" : {
+                        "total" : {
+                          "value" : 1,
+                          "relation" : "eq"
+                        },
+                        "max_score" : 0.6931471,
+                        "hits" : [
+                          {
+                            "_index" : "my_index",
+                            "_id" : "2",
+                            "_nested" : {
+                              "field" : "comments",
+                              "offset" : 0
+                            },
+                            "_score" : 0.6931471,
+                            "_source" : {
+                              "message" : "This is a comment",
+                              "user" : "John"
+                            }
+                          }
+                        ]
+                      }
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+You can also use inner_hits with following features:
+
+  - [Highlight query matches]({{site.url}}{{site.baseurl}}/search-plugins/searching-data/highlight/)
+  - [Explain]({{site.url}}{{site.baseurl}}/api-reference/explain/)
+
+<!-- vale off -->
+## Inner_hits parameters
+<!-- vale on -->
+There are additional parameters that can be passed to search with `inner_hits` using both nested objects and parent-join relationships in OpenSearch. Such as `from`, `size`, `sort` and `name`.
+
+* `from`: The offset from where to start fetching hits within the inner_hits results.
+* `size`: The maximum number of inner hits to return.
+* `sort`: The sorting order for the inner hits.
+* `name`: A custom name for the inner hits in the response. This is useful to differentiate between multiple inner hits in a single query.
+<!-- vale off -->
+### Example of inner_hits parameters with nested objects
+<!-- vale on -->
+1. Create index and mappings:
+
+    ```json
+    PUT /products
+    {
+      "mappings": {
+        "properties": {
+          "product_name": { "type": "text" },
+          "reviews": {
+            "type": "nested",
+            "properties": {
+              "user": { "type": "text" },
+              "comment": { "type": "text" },
+              "rating": { "type": "integer" }
+            }
+          }
+        }
+      }
+    }
+    ```
+    {% include copy-curl.html %}
+
+2. Index data:
+
+    ```json
```

Comment:
Please separate the indexed documents into one code block each and add a copy-curl include after each one. Please apply globally.